src/main.c

```diff
@@ -1220,15 +1222,51 @@ static int on_config_max_connections(h2o_configurator_command_t *cmd, h2o_config
     return h2o_configurator_scanf(cmd, node, "%d", &conf.max_connections);
 }
 
+static void add_one_to_cpu_list(int **cpu_list, size_t *num_threads, int cpu)
+{
+    *num_threads += 1;
+    (*cpu_list) = realloc(*cpu_list, sizeof(**cpu_list) * (*num_threads + 1));
+    (*cpu_list)[*num_threads - 1] = cpu;
+    (*cpu_list)[*num_threads] = -1;
```

Comment:
Are you concerned about having overruns here?

I ask this because, *if* you are concerned about that, *then* I might suggest doing a refactor. Replace `conf.num_threads` and `conf.cpu_list` with a `H2O_VECTOR(int)`. The size of the vector indicates the number of threads, and each value of the vector indicates the CPU id (if non-negative) to be pinned to, or if it's not pinned (if -1). By taking that approach, there would be no fear of losing synchronization between `num_threads` and `cpu_list`, and you'd also be able to use things like `h2o_vector_reserve`.

I am not saying that we should make such change, but it's generally the case that I prefer avoiding having extra complexity for making the code "fail-safe."